### PR TITLE
vandheer keymap of satan

### PR DESCRIPTION
**Pull request by phluan**
## Types of changes

- [ ] Core
- [ ] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by this PR

* 

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mechmerlin** [comment]
You left the description section blank, could you explain what this PR is for?

There already exists a GH60 Satan PCB. Is this a handwire that uses the same pins and has the same features? If so, what is the purpose of this PR?

File: keyboards/satan/keymaps/vandheer/keymap.c [hunk 1]
```diff
@@ -0,0 +1,210 @@
+#include "satan.h"
```
Comment (mechmerlin) [inline keyboards/satan/keymaps/vandheer/keymap.c]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/satan/keymaps/vandheer/keymap_template.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+   * |     |   |   |   |   |   |   |   |   |   |   |   |   |     |
+   * |-----------------------------------------------------------|
+   * |Nav~Esc|   |   |   |   |   |   |   |   |   |   |   |       |
+   * |-----------------------------------------------------------|
+   * |Shift   |   |   |   |   |   |   |   |   |   |   |          |
+   * |-----------------------------------------------------------|
+   * |Ctrl|Alt |Gui |                        |Gui |Alt |Fn |Macro|
+   * `-----------------------------------------------------------'
+   */
+[_BLANK] = KEYMAP_ANSI(
```
Comment (mechmerlin) [inline keyboards/satan/keymaps/vandheer/keymap_template.c]:
```suggestion
[_BLANK] = LAYOUT_ansi(
```

File: keyboards/satan/keymaps/vandheer/rules.mk [hunk 3]
```diff
@@ -0,0 +1,70 @@
+
+# MCU name
+#MCU = at90usb1287
```
Comment (mechmerlin) [inline keyboards/satan/keymaps/vandheer/rules.mk]:
```suggestion

```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]: If you could replace the include guard (the ifndef, define, and endif at the end) with just `#pragma once`? 

It's simpler, and less prone to user error, so if you could make this switch? 

```suggestion
#pragma once
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]: This line is not actually needed, and should be removed. 

Specifically, this should be included in the keyboard's `config.h` file, and is redundant. 

```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]: You do not need all of this.  This is for the keyboard.  

You only need the content that is not in the keyboard's config.h. 

```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+#define PRODUCT_ID      0x6060
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+#define DEVICE_VER      0x0003
+#define MANUFACTURER    SATAN
+#define PRODUCT         GH60
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+#ifndef CONFIG_H
+#define CONFIG_H
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0003
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+#define MANUFACTURER    SATAN
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+#define PRODUCT         GH60
+#define DESCRIPTION     QMK keyboard firmware for Satan GH60 with WS2812 support
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+/* key matrix size */
+#define MATRIX_ROWS 5
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+#define MATRIX_COLS 14
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+
+// ROWS: Top to bottom, COLS: Left to right
+
+#define MATRIX_ROW_PINS { D0, D1, D2, D3, D5 }
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+#define MATRIX_COL_PINS { F0, F1, E6, C7, C6, B7, D4, B1, B0, B5, B4, D7, D6, B3 }
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+#define UNUSED_PINS
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+
+#define BACKLIGHT_PIN B6
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+
+/* Backlight configuration
+ */
+#define BACKLIGHT_LEVELS 4
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+
+/* Underlight configuration
+ */
+
+#define RGB_DI_PIN E2
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLIGHT_TIMER
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLED_NUM 8     // Number of LEDs
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLIGHT_HUE_STEP 10
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 24]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLIGHT_SAT_STEP 17
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 25]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLIGHT_VAL_STEP 17
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 26]
```diff
@@ -0,0 +1,139 @@
...
+
+/* Enable Backlight Breathing */
+#define BACKLIGHT_BREATHING
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 27]
```diff
@@ -0,0 +1,139 @@
...
+#define BREATHING_PERIOD 6
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 28]
```diff
@@ -0,0 +1,139 @@
...
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 29]
```diff
@@ -0,0 +1,139 @@
...
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 30]
```diff
@@ -0,0 +1,139 @@
...
+
+// Waiting time for next tap dance to registered
+#define TAPPING_TERM 200
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 31]
```diff
@@ -0,0 +1,139 @@
...
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 32]
```diff
@@ -0,0 +1,139 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 33]
```diff
@@ -0,0 +1,139 @@
...
+
+/* key combination for command */
+#define IS_COMMAND() ( \
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 34]
```diff
@@ -0,0 +1,139 @@
...
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 35]
```diff
@@ -0,0 +1,139 @@
...
+)
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 36]
```diff
@@ -0,0 +1,139 @@
...
+
+/* Underlight configuration
+ */
+
+#define RGB_DI_PIN E2
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 37]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLIGHT_TIMER
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 38]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLED_NUM 8     // Number of LEDs
+#define RGBLIGHT_HUE_STEP 10
+#define RGBLIGHT_SAT_STEP 17
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 39]
```diff
@@ -0,0 +1,139 @@
...
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* Underlight configuration
+ */
+
+#define RGB_DI_PIN E2
+#define RGBLIGHT_TIMER
+#define RGBLED_NUM 8     // Number of LEDs
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 40]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLIGHT_HUE_STEP 10
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/config.h [hunk 41]
```diff
@@ -0,0 +1,139 @@
...
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
+
+#endif
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/config.h]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/keymap.c [hunk 42]
```diff
@@ -0,0 +1,210 @@
...
+
+const uint16_t PROGMEM fn_actions[] = {
+  [0]  = ACTION_FUNCTION(GRAVE_ON_ESC),
+  [1]  = ACTION_FUNCTION(GRAVE_ON_TAB)
+};
+
+void action_function(keyrecord_t *record, uint8_t id, uint8_t opt) {
+  static uint8_t shift_esc_shift_mask;
+  switch (id) {
+    case GRAVE_ON_ESC:
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/keymap.c]: This is covered by `KC_GESC`

File: keyboards/satan/keymaps/vandheer/keymap.c [hunk 43]
```diff
@@ -0,0 +1,210 @@
...
+  __n__,  __n__,  __n__,  __n__,  __n__,  __n__,  __n__,  __n__,  __n__,  __n__,  __n__,                  __n__, \
+  __n__,  __n__,  __n__,                           __n__,                         __n__,  __n__,  __n__,  _______),
+};
+
+enum function_id {
+    GRAVE_ON_ESC,
+    GRAVE_ON_TAB,
+};
+
+const uint16_t PROGMEM fn_actions[] = {
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/keymap.c]: Could you remove the `fn_actions` block?  All of it's features have been added as keycodes, and been deprecated in favor of `process_record_user`.

File: keyboards/satan/keymaps/vandheer/keymap.c [hunk 44]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+// Tap Dance Definitions
+qk_tap_dance_action_t tap_dance_actions[] = {
+  //Tap once for Esc, twice for Caps Lock
+  [TD_SPC_ENT]  = ACTION_TAP_DANCE_DOUBLE(KC_SPC, KC_ENT)
+};
+
+// Macro
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt) // this is the function signature -- just copy/paste it into your keymap file as it is.
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/keymap.c]: Could you remove the `action_get_macro` function?  It's been deprecated in favor of `process_record_user`. 
https://docs.qmk.fm/#/feature_macros

File: keyboards/satan/keymaps/vandheer/rules.mk [hunk 45]
```diff
@@ -0,0 +1,70 @@
...
+#   Atmel DFU loader 4096
+#   LUFA bootloader  4096
+#   USBaspLoader     2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no	# Virtual DIP switch configuration(+1000)
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/rules.mk]: This block is all that you need.  The rest of the file should be removed.

File: keyboards/satan/keymaps/vandheer/rules.mk [hunk 46]
```diff
@@ -0,0 +1,70 @@
...
+RGBLIGHT_ENABLE = no   # Enable keyboard underlight functionality (+4870)
+BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality (+1150)
+MIDI_ENABLE = no 		# MIDI controls
+AUDIO_ENABLE = no
+UNICODE_ENABLE = no 		# Unicode
+BLUETOOTH_ENABLE = no # Enable Bluetooth with the Adafruit EZ-Key HID
+TAP_DANCE_ENABLE = yes	# Enable tap dance feature
+SLEEP_LED_ENABLE ?= no # Enables your LED to breath while your computer is sleeping. Timer1 is being used here. This feature is largely unused and untested, and needs updating/abstracting.
+
+LAYOUTS = 60_ansi 60_iso 60_ansi_split_bs_rshift
```
Comment (drashna) [inline keyboards/satan/keymaps/vandheer/rules.mk]:
```suggestion
```

File: keyboards/satan/keymaps/vandheer/keymap_template.c (hunk 2 above)
Comment (noroadsleft) [reply keyboards/satan/keymaps/vandheer/keymap_template.c]: Actually, this should be:

```suggestion
[_BLANK] = LAYOUT_60_ansi(
```

**Comment by drashna** [comment]
Just wanted to touch base here.

Any update on this?
Or is it okay if we close this?

**Comment by drashna** [comment]
It's been more than 8 months with any updates or comments on this PR.  

I'm closing the PR.   If you want to update this, feel free to re-open the PR or open a new one

**Pull request closed without merging**